### PR TITLE
tstruct takes kwargs

### PR DESCRIPTION
**Pull request by tpoterba**

### CODE REVIEW

File: python/hail/expr/aggregators.py [hunk 1]
```diff
@@ -569,8 +569,12 @@ def stats(expr):
     agg = _to_agg(expr)
     if not is_numeric(agg._type):
         raise TypeError("'stats' expects a numeric argument, found '{}'".format(agg._type))
-    return _agg_func('stats', agg, tstruct(['mean', 'stdev', 'min', 'max', 'nNotMissing', 'sum'],
-                                           [tfloat64, tfloat64, tfloat64, tfloat64, tint64, tfloat64]))
+    return _agg_func('stats', agg, tstruct(mean=tfloat64,
+                                           stdev=tfloat64,
+                                           min=tfloat64,
+                                           max=tfloat64,
+                                           nNotMissing=tint64,
```
Comment (jbloom22) [inline python/hail/expr/aggregators.py]: `n_present > n_not_missing > nNotMissing` but perhaps outside scope

File: python/hail/expr/aggregators.py [hunk 2]
```diff
@@ -1022,6 +1031,8 @@ def hist(expr, start, end, bins):
     agg = _to_agg(expr)
     if not is_numeric(agg._type):
         raise TypeError("'hist' expects argument 'expr' to be a numeric type, found '{}'".format(agg._type))
-    t = tstruct(['binEdges', 'binFrequencies', 'nLess', 'nGreater'],
-                [tarray(tfloat64), tarray(tint64), tint64, tint64])
+    t = tstruct(binEdges=tarray(tfloat64),
```
Comment (jbloom22) [inline python/hail/expr/aggregators.py]: names are not pythonic, i see the same throughout functions, so probably best to note that something needs to tackle this for 0.2 in another PR

File: python/hail/expr/types.py [hunk 3]
```diff
@@ -617,16 +589,46 @@ def fields(self):
...
+        ts._get_jtype = lambda: scala_object(Env.hail().expr.types, 'TStruct').apply([f.name for f in fields],
+                                                                                     [f.dtype._jtype for f in fields],
+                                                                                     False)
+        super(TStruct, ts).__init__()
+        return ts
+
+    @staticmethod
+    @typecheck(names=listof(str), types=listof(Type))
+    def from_lists(names, types):
+        """Construct a :class:`.TStruct` from lists of names and types.
```
Comment (jbloom22) [inline python/hail/expr/types.py]: "from a list of names and a list of corresponding types."

File: python/hail/expr/types.py [hunk 4]
```diff
@@ -617,16 +589,46 @@ def fields(self):
...
+
+        Parameters
+        ----------
+        names : :obj:`list` of :obj:`str`
+            Field names.
+        types : :obj:`list` of :class:`.Type`
```
Comment (jbloom22) [inline python/hail/expr/types.py]: Add "Field types." on next line

File: python/hail/expr/types.py [hunk 5]
```diff
@@ -617,16 +589,46 @@ def fields(self):
...
 
-        :return: TStruct from input fields
-        :rtype: :class:`.TStruct`
+        Returns
+        -------
+        :class:`.TStruct`
         """
-        return TStruct([f.name for f in fields], [f.dtype for f in fields])
+        if not len(names) == len(types):
+            raise ValueError("'from_lists': parameters 'names' and 'types' had different lengths ({} / {})"
```
Comment (jbloom22) [inline python/hail/expr/types.py]: had => have

Comment (jbloom22) [reply python/hail/expr/types.py]: And I'd write "different lengths: {}, {}"

File: python/hail/tests.py [hunk 6]
```diff
@@ -197,8 +197,8 @@ def test_select(self):
         self.assertEqual(set(kt.select(kt.a, foo=kt.a + kt.b - kt.c - kt.d, **kt.g).columns), {'a', 'foo', 'x', 'y'})
 
     def test_aggregate(self):
-        schema = hl.tstruct(['status', 'GT', 'qPheno'],
-                            [hl.tint32, hl.tcall, hl.tint32])
+        schema = hl.tstruct.from_lists(['status', 'GT', 'qPheno'],
```
Comment (jbloom22) [inline python/hail/tests.py]: we use `quant_pheno` elsewhere, might as well be consistent / pythonic

File: python/hail/tests.py [hunk 7]
```diff
@@ -237,8 +237,8 @@ def test_aggregate(self):
         self.assertDictEqual(result, expected)
 
     def test_errors(self):
-        schema = hl.tstruct(['status', 'gt', 'qPheno'],
-                            [hl.tint32, hl.tcall, hl.tint32])
+        schema = hl.tstruct.from_lists(['status', 'gt', 'qPheno'],
```
Comment (jbloom22) [inline python/hail/tests.py]: same

File: python/hail/tests.py [hunk 8]
```diff
@@ -334,52 +334,45 @@ def test_from_spark_works(self):
...
         vds = vds.annotate_cols(y1=agg.count(),
                                 y2=agg.fraction(False),
                                 y3=agg.count_where(True),
                                 y4=vds.foo + vds.apple)
 
-        expected_schema = hl.tstruct(['s', 'apple', 'y1', 'y2', 'y3', 'y4'],
-                                     [hl.tstr,
-                                      hl.tint32, hl.tint64, hl.tfloat64, hl.tint64, hl.tint32])
+        expected_schema = hl.tstruct.from_lists(['s', 'apple', 'y1', 'y2', 'y3', 'y4'],
+                                                [hl.tstr,
```
Comment (jbloom22) [inline python/hail/tests.py]: make one line

File: python/hail/tests.py [hunk 9]
```diff
@@ -698,17 +691,20 @@ def test_range(self):
         ds = hl.utils.range_matrix_table(100, 10)
         self.assertEqual(ds.count_rows(), 100)
         self.assertEqual(ds.count_cols(), 10)
-        et = ds.annotate_entries(entry_idx = 10 * ds.row_idx + ds.col_idx).entries_table().index()
+        et = ds.annotate_entries(entry_idx=10 * ds.row_idx + ds.col_idx).entries_table().index()
```
Comment (jbloom22) [inline python/hail/tests.py]: really? space even when rhs is a longer expression? looks weird  to me

File: python/hail/expr/aggregators.py (hunk 1 above)
Comment (tpoterba) [reply python/hail/expr/aggregators.py]: definitely outside scope. This involves scala changes.

File: python/hail/tests.py (hunk 6 above)
Comment (tpoterba) [reply python/hail/tests.py]: orthogonal.

File: python/hail/tests.py (hunk 9 above)
Comment (tpoterba) [reply python/hail/tests.py]: this was an autoformat change, I reverted it.

**Pull request merged**